File: passwordStrengthChecker/passwordStrengthChecker.go
package main

import (
	"log"
	"os"
	"strings"
)

func main() {
    args := os.Args
    if len(args) == 1 {
        log.Fatal("Specify a password")
    }
    if len(args[1]) < 9 {
        log.Println("Password is so short")
    } else if !(strings.IndexFunc(args[1], letters) >= 0){
        log.Println("Password must containt lower case (at least one)")
    }  else if !(strings.IndexFunc(args[1], capitals) >= 0){
        log.Println("Password must containt upper case (at least one)")
    } else if !strings.ContainsAny(args[1], "!@#$%&*") {
        log.Println("Password must contain one of these '!@#$%&*'")
    } else {
        log.Println("Password is perfecto")
    }
}

func letters(r rune) bool {
        if (r >= 97 && r <= 122) {
            return true
        }
        return false
    }

func capitals(r rune) bool {
    if (r >= 65 && r <= 90) {
        return true
    } else { return false }
}

File: passwordStrengthChecker/passwordStrengthChecker.py
from sys import argv

def contiansUpper(password: str):
    for digit in password:
        if digit > "A" and digit < "Z":
            return True
    return False

def contiansLower(password: str):
    for digit in password:
        if digit > "a" and digit < "z":
            return True
    return False

def containSpecial(passowrd: str):
    for digit in passowrd:
        if digit in "!@#$%&*":
            return True
    return False

if len(argv) <= 1:
    print("Append a password when calling a script")

password = argv[1]


if password.__len__() < 8:
    print("password is to short, must be 8 digits long")
elif contiansLower(password):
    print("password must contain at least one lower case")
elif contiansUpper(password):
    print("password must contain at least one upper case")
elif not containSpecial(password):
    print("password must contain at least one of these '!@#$%&*'")
else:
    print("password is perfecto")

